opm/simulators/linalg/PropertyTree.hpp

```diff
@@ -50,14 +50,19 @@ class PropertyTree {
     template<class T>
     T get(const std::string& key, const T& defValue) const;
 
+  //PropertyTree& get_child(const std::string& key) const;
+  
     PropertyTree get_child(const std::string& key) const;
 
     std::optional<PropertyTree> get_child_optional(const std::string& key) const;
 
     PropertyTree& operator=(const PropertyTree& tree);
 
     void write_json(std::ostream& os, bool pretty) const;
-
+    boost::property_tree::ptree* getBoostParamPtr() const{
+      //just to be able to use full boost interface
+      return tree_.get();
+    }
```

Comment:
this commit doesn't seem ready for inclusion in main line. if we need access to the whole api in simulator code the whole approach of encapsulation using this class falls on its rear end and we should consider removing the encapsulation.

Reply:
I agree that this breaks encapsulation. I need a way to put values back into the tree, and especially into a subtree, and this was the quickest way to go about doing that.  In particular, I didn't see a way of combining [`get_child()`](https://github.com/OPM/opm-simulators/blob/032b296836fff00647d686d12d154609ae7474f2/opm/simulators/linalg/PropertyTree.hpp#L53) with `put<>()` in a way that would make the change persistent.  I am open to suggestions.

Reply:
https://github.com/akva2/opm-simulators/commit/0106bc9842434430e49c0e4e154e6e1c533507eb

should in principle do it, although it's has some rather nasty not-so-const-correct aspects.

Reply:
> [akva2@0106bc9](https://github.com/akva2/opm-simulators/commit/0106bc9842434430e49c0e4e154e6e1c533507eb)
> 
> should in principle do it, although it's has some rather nasty not-so-const-correct aspects.

Okay, thanks.  I'll pick this back up after the holidays, then.